model/value.go

```diff
@@ -414,3 +379,8 @@ func (mat Matrix) String() string {
 
 	return strings.Join(strs, "\n")
 }
+
+type GenericSamplePair interface {
```

Comment:
Is this used anywhere?

Reply:
We use it in Mimir to reduce code duplication, since many functions to process SamplePair and SampleHistogramPair rely only on the timestamp and can be combined into one generic function instead of having 2 separate functions with essentially the same code. Unfortunately Golang doesn't allow us to define such generics for imported types, so we can't do this in Mimir. I understand the need to support 1.17 here for now though, so I've reverted it. I was thinking since the next release is likely to be in the next few months and then (maybe) we can drop 1.17, we could keep the generics and the version bump in a separate branch which we could use in Mimir, and merge only after it's safe to drop 1.17. Even after reverting though, the CI test is failing for only 1.17, any idea why?

Reply:
Test failure seems to be flaky. I re-ran it and now it works.